### PR TITLE
refactor: move some functions and module-level state into classes as private methods and properties to start to encapsulate Docsify

### PR DESCRIPTION
**Pull request by trusktr**
## **Summary**

Also some small tweaks:

- move initGlobalAPI out of Docsify.js to start to encapsulate Docsify
- move ajax to utils folder
- fix some type definitions and improve content in some JSDoc comments
- use concise class field syntax in some spots
- consolidate duplicate docsify-ignore comment removal code

## Related issues

This handles a task in

-  https://github.com/docsifyjs/docsify/issues/2104

as well as works towards

- https://github.com/docsifyjs/docsify/issues/2135

Continues from PR

- https://github.com/docsifyjs/docsify/pull/2134

## **What kind of change does this PR introduce?**

  Code style update
  Refactor

## **For any code change,**

- [x] Related documentation has been updated if needed
- [x] Related tests have been updated or tests have been added

## **Does this PR introduce a breaking change?** (check one)

No

If yes, please describe the impact and migration path for existing applications:

## **Tested in the following browsers:**

- [x] Chrome
- [x] Firefox
- [x] Safari
- [x] Edge

### CODE REVIEW

File: src/core/event/index.js [hunk 1]
```diff
@@ -18,29 +20,300 @@ export function Events(Base) {
...
+
+    /** @readonly */
+    #nav = {};
+
+    #hoverOver = false;
+    #scroller = null;
+    #enableScrollEvent = true;
+    #coverHeight = 0;
+
+    #scrollTo(el, offset = 0) {
```
Comment (trusktr) [inline src/core/event/index.js:56]: logic was not changed, only moved.

File: src/core/fetch/index.js [hunk 2]
```diff
@@ -1,70 +1,70 @@
...
+/** @typedef {import('../Docsify.js').Constructor} Constructor */
 
 /**
  * @template {!Constructor} T
  * @param {T} Base - The class to extend
  */
 export function Fetch(Base) {
-  let last;
+  return class Fetch extends Base {
+    #loadNested(path, qs, file, next, vm, first) {
```
Comment (trusktr) [inline src/core/fetch/index.js:14]: logic was not changed, only moved.

File: src/core/event/index.js [hunk 3]
```diff
@@ -18,29 +20,300 @@ export function Events(Base) {
...
+      dom.on(sidebar, 'mouseover', () => {
+        this.#hoverOver = true;
+      });
+      dom.on(sidebar, 'mouseleave', () => {
+        this.#hoverOver = false;
+      });
+    }
+
+    #scrollIntoView(path, id) {
+      if (!id) {
```
Comment (Koooooo-7) [inline src/core/event/index.js:189]: what does the `#functionName` means here? a private one?

File: src/core/event/index.js [hunk 4]
```diff
@@ -18,29 +20,300 @@ export function Events(Base) {
...
       if (this.config.coverpage) {
-        !isMobile && on('scroll', sidebar.sticky);
+        !isMobile && on('scroll', this.__sticky);
       } else {
         body.classList.add('sticky');
       }
     }
+
+    /** @readonly */
+    #nav = {};
```
Comment (Koooooo-7) [inline src/core/event/index.js:49]: what does the `#param` means here? a private one? should be `_param`?

**Review by Koooooo-7 — COMMENTED** [review]
should we change all the `_private` stuff to `#private` into ES6? otherwise, it seems messy here with both format.

**Comment by Koooooo-7** [comment]
Closed since no changes detect (weird).
image

**Pull request closed without merging**